### PR TITLE
Yamux parser improvements

### PR DESCRIPTION
**Pull request by AJIOB**
* Adding more precise fields decoding
* Adding automatic window size calculations
* Fixing comments typos

### CODE REVIEW

File: src/yamux/yamux.lua [hunk 1]
```diff
@@ -355,6 +437,29 @@ local function parse_yamux(subtree, pinfo, fields, yamux_header)
         set_protocol(pinfo)
     end
 
+    if is_yamux_stream_default_created() and not (typeValue == hdr_type.PING or typeValue == hdr_type.GO_AWAY) then
+        dprint("Default opened stread cannot be with type", typeValue)
```
Comment (blake) [inline src/yamux/yamux.lua]:
```suggestion
        dprint("Default opened stream cannot be with type", typeValue)
```
Did you mean stream?

File: src/yamux/yamux.lua [hunk 2]
```diff
@@ -411,6 +516,40 @@ local function parse_yamux(subtree, pinfo, fields, yamux_header)
...
+    if not (disable_window_calc_logic or is_yamux_stream_default_created()) then
+        -- Window values recalculation
+        local new_server_window_size = window_start_value
+        local new_client_window_size = window_start_value
+
+        if previous_frame then
+            -- Use prev values, if prev frame is presented
+            new_server_window_size = yamux_stream_info[current_stream_id].server_window_size
+            new_client_window_size = yamux_stream_info[current_stream_id].client_window_size
+        end
```
Comment (blake) [inline src/yamux/yamux.lua]:
```suggestion
        if previous_frame and (yamux_stream_info[current_stream_id].client_window_size and yamux_stream_info[current_stream_id].server_window_size) then
            -- Use prev values, if prev frame is presented
            new_server_window_size = yamux_stream_info[current_stream_id].server_window_size
            new_client_window_size = yamux_stream_info[current_stream_id].client_window_size
        end
```
This assumes that the `client_window_size` and `server_window_size` variables have already been initialized for the stream.

When I ran the tests in this repo, I saw that it is possible with a partial packet capture that the dissector is able to correlate the frame to an existing stream that had not yet had the window size variables initialized. I suggest extending the conditional to ensure that the variables are not nil before trying to set them as the new window size.

File: src/yamux/yamux.lua [hunk 3]
```diff
@@ -355,6 +437,29 @@ local function parse_yamux(subtree, pinfo, fields, yamux_header)
...
+
+    -- Determine client/server tcp ports for window calculation
+    local disable_window_calc_logic = false
+    if flags == HDR_FLAG_NAME_MAP.SYN then
+        if src_port().value == dst_port().value then
+            disable_window_calc_logic = true
+            dprint("TCP source port", src_port().value, "is equal to destination port", dst_port().value, ". No possibility to determine Yamux client/server direction. Window calculation logic will be disabled")
+        elseif is_yamux_stream_default_created() then
+            disable_window_calc_logic = true
+            dprint("Default opened stread cannot be opened")
```
Comment (blake) [inline src/yamux/yamux.lua]:
```suggestion
            dprint("Default opened stream cannot be opened")
```

**Review by blake — COMMENTED** [review]
Looks good. Thanks for adding the window size calculation logic.

I left a couple of suggestions. I also recommend updating the test data to reflect the fact that the dissector can now parse these new fields. You can find details on how to do that in <https://github.com/blake/wireshark-consul-dissector/blob/main/tests/README.md>.

tl;dr is:

```shell
$ luarocks install luaunit
$ GENERATE_PROTOCOL_DATA=1 lua test_decode.lua
```

**Comment by AJIOB** [comment]
Hi Blake.

Sorry for a long fixing.

I've fixed all what I want & improve Yamux client-server direction detection.

I think it can be merged, if all is OK.

**Review by blake — APPROVED** [review]
Thanks for making those changes.

I made one small change to remove the debug output from the Yamux protocol data.  I'll go ahead and merge this PR now.
